### DISCUSSION TITLE
Why cant I access AR mode

### DISCUSSION THREAD

**Discussion by hmejbri** [pmndrs/xr#327] [Q&A] [answered]
I followed every step in the documentation and I still cant access the AR mode in my phone (It supports webXR). and when I use the XR emulator extension I dont see my object anywhere.

This is my code :

Model.jsx 
```
import React from "react";
import { useGLTF } from "@react-three/drei";
import { DRACOLoader } from "three-stdlib";

const Model = () => {
  const { scene } = useGLTF("/models/villa/villasss.gltf", true, true, (loader) => {
    // Draco loader setup
    loader.setDRACOLoader(new DRACOLoader().setDecoderPath("/models/villa"));
  });

  scene.scale.set(0.1, 0.1, 0.1); // Example: scaling down the model
  scene.position.set(-0.2, -0.2, -0.2);
  return <primitive object={scene} />;
};

export default Model;
```

ModelContainer.jsx 
```
import React, { Suspense, useEffect, useRef, useState } from "react";
import Model from "./Model";
import { DirectionalLightHelper } from "three";
import { Html } from "@react-three/drei";
import { Canvas } from "@react-three/fiber";
import Controls from "../Controls/Controls";
import { XR, createXRStore } from '@react-three/xr';

const ModelContainer = () => {
  const store = createXRStore();
  const dirLightRef = useRef();

  return (
    <>
      <div className="bottom-right-container">
        <button onClick={() => store.enterAR()}><img src="icons/AR.png" width="40" height="40" /></button>
        <button onClick={() => store.enterVR()}><img src="icons/VR.png" width="40" height="40" /></button>
      </div>

      <Canvas camera={{ position: [0, 1, 0] }}>
        <XR store={store}>
          <Controls />
          <ambientLight intensity={2} />
          <directionalLight position={[1, 1, 0]} intensity={3} ref={dirLightRef} />
          <Suspense
            fallback={
              <Html center>
                <div className="loading-container">
                  Loading
                </div>
              </Html>
            }
          >
            <Model />
          </Suspense>
        </XR>
      </Canvas>
    </>
  );
};

export default ModelContainer;

```

**Comment by bbohlender** [comment]
@hmejbri
2 things I can see that are incorrect:
- the suspense fallback is html, which cannot be rendered in WebXR. Only in handheld ar you can use XRDomOverlay
- you are executing createXRStore every time the component re-renders -> either put it outside of the component or wrap it in a useMemo

**Reply by hmejbri** [reply]
Thank you for your response, I fixed those 2 problems but still cant access AR. After further investigation I can see the model using the emulator extension but Its behind the camera.

**Reply by bbohlender** [reply]
what exactly happens if you try to access AR?

**Reply by hmejbri** [reply]
Nothing. 

I'm working on a remote server not on localhost and its http not https, can that be the reason ?

**Reply by bbohlender** [reply, marked as answer]
yes
https://docs.pmnd.rs/xr/advanced/pitfalls#missing-https

**Reply by bbohlender** [reply]
here's a exmplanation of a recommended dev setup
https://docs.pmnd.rs/xr/getting-started/development-setup

**Reply by decipher-cs** [reply]
> yes https://docs.pmnd.rs/xr/advanced/pitfalls#missing-https

That link no longer works but it's probably referencing to [this](https://pmndrs.github.io/xr/docs/getting-started/faq#i-cannot-enter-the-xr-session!)
